core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -581,7 +583,12 @@ object Battle {
         }
 
         if (!thisCombatant.isDefeated() && !unitCouldAlreadyPromote && promotions.canBePromoted())
-            civ.addNotification("[${thisCombatant.unit.displayName()}] can be promoted!",thisCombatant.getTile().position, NotificationCategory.Units, thisCombatant.unit.name)
+            thisCombatant.run {
```

Comment:
I really don't see the need to put this in a 'run' scope, it would be clearer without...